x/coinswap/client/rest/tx.go

```diff
@@ -0,0 +1,165 @@
+		Deposit       sdk.Coin       `json:"deposit"`
+		DepositAmount sdk.Int        `json:"deposit_amount"`
+		MinReward     sdk.Int        `json:"min_reward"`
+		Deadline      time.Time      `json:"deadline"`
+		Sender        sdk.AccAddress `json:"sender"` // in bech32
+	}
+
+	// RemoveLiquidityRequest defines the properties of a remove liquidity request's body.
+	RemoveLiquidityRequest struct {
+		BaseReq        rest.BaseReq   `json:"base_req"`
+		Withdraw       sdk.Coin       `json:"withdraw"`
+		WithdrawAmount sdk.Int        `json:"withdraw_amount"`
+		MinNative      sdk.Int        `json:"min_native"`
+		Deadline       time.Time      `json:"deadline"`
+		Sender         sdk.AccAddress `json:"sender"` // in bech32
+	}
+
+	// SwapOrderRequest defines the properties of a swap order request's body.
+	SwapOrderRequest struct {
+		BaseReq    rest.BaseReq   `json:"base_req"`
+		Input      sdk.Coin       `json:"input"`
+		Output     sdk.Coin       `json:"output"`
+		Deadline   time.Time      `json:"deadline"`
+		Sender     sdk.AccAddress `json:"sender"`    // in bech32
+		Recipient  sdk.AccAddress `json:"recipient"` // in bech32
+		IsBuyOrder bool           `json:"is_buy_order"`
+	}
+)
+
+func postAddLiquidityHandlerFn(cliCtx context.CLIContext) http.HandlerFunc {
+	return func(w http.ResponseWriter, r *http.Request) {
+		var req AddLiquidityRequest
+
+		if !rest.ReadRESTReq(w, r, cliCtx.Codec, &req) {
+			return
+		}
+
+		req.BaseReq = req.BaseReq.Sanitize()
+		if !req.BaseReq.ValidateBasic(w) {
+			return
+		}
+
+		msg := types.NewMsgAddLiquidity(req.Deposit, req.DepositAmount, req.MinReward, req.Deadline, req.Sender)
+		if err := msg.ValidateBasic(); err != nil {
+			rest.WriteErrorResponse(w, http.StatusBadRequest, err.Error())
+			return
+		}
+
+		fromAddr, err := sdk.AccAddressFromBech32(req.BaseReq.From)
+		if err != nil {
+			rest.WriteErrorResponse(w, http.StatusBadRequest, err.Error())
+			return
+		}
+
+		if !bytes.Equal(fromAddr, req.Sender) {
+			rest.WriteErrorResponse(w, http.StatusUnauthorized, "must use own sender address")
+			return
+		}
+
+		utils.WriteGenerateStdTxResponse(w, cliCtx, req.BaseReq, []sdk.Msg{msg})
+	}
+}
+
+func postRemoveLiquidityHandlerFn(cliCtx context.CLIContext) http.HandlerFunc {
+	return func(w http.ResponseWriter, r *http.Request) {
+		var req RemoveLiquidityRequest
+
+		if !rest.ReadRESTReq(w, r, cliCtx.Codec, &req) {
+			return
+		}
+
+		req.BaseReq = req.BaseReq.Sanitize()
+		if !req.BaseReq.ValidateBasic(w) {
+			return
+		}
+
+		msg := types.NewMsgRemoveLiquidity(req.Withdraw, req.WithdrawAmount, req.MinNative, req.Deadline, req.Sender)
+		if err := msg.ValidateBasic(); err != nil {
+			rest.WriteErrorResponse(w, http.StatusBadRequest, err.Error())
+			return
+		}
+
+		fromAddr, err := sdk.AccAddressFromBech32(req.BaseReq.From)
+		if err != nil {
+			rest.WriteErrorResponse(w, http.StatusBadRequest, err.Error())
+			return
+		}
+
+		if !bytes.Equal(fromAddr, req.Sender) {
```

Comment:
this seems redundant to me. I know this is done in other places of the SDK, does anyone know why? I don't see why we can't just get rid of the req.Sender field

Reply:
@colin-axner Since we don't need the `Sender` in the requests, I think we should remove it from the msg structs too.

Reply:
We need it for `GetSigners` func

Reply:
bump for removing req.Sender, use the fromAddr for the sender field in Msg